### ISSUE TITLE
withResponseType fails to distinguish bounded types (e.g., T extends MyClass)

### ISSUE THREAD

**Issue by landonreed** [manusant/spark-swagger#14]
In `DefinitionsFactory#getCollectionType`, the code that determines the type for a collection field fails to recognize a [bounded type](https://docs.oracle.com/javase/tutorial/java/generics/bounded.html), e.g., `MyClass` in `List<T extends MyClass>`.

https://github.com/manusant/spark-swagger/blob/f7c0c8ef3554b43afc1771ea90ced01a50bc7125/src/main/java/io/github/manusant/ss/factory/DefinitionsFactory.java#L169-L180

Some additional checks that follow the reflection/type checks shown above might be useful in determining the type.

There is also a somewhat related issue here in line 180. A failure to decode the type will result in a default return value of `String.class`, but a more appropriate default might be `Object.class`.

**Comment by landonreed** [comment]
@manusant, apologies for not updating this PR, but I'm not sure that my changes actually resolve #14. I had trouble testing these changes out without importing it into a project of mine that uses spark-swagger as a dep. 

tldr; This might need more investigation before closing out the referenced issue.

**Comment by manusant** [comment]
In this case you should use **withResponseAsCollection** method to specify the typed collection. I decided to keep it simple when it comes to typed objects. Unfortunatelly i had little to no time last year to take proper care of this project but in the biginning of next year i'm planning some improvements and support for swagger 3.

**Comment by manusant** [comment]
Contributions will be wellcome and that's why i want to move to github packages and actions to facilitate contributions
